math/mathcore/src/FitUtil.cxx

```diff
@@ -638,7 +638,8 @@ double FitUtil::EvaluateChi2Residual(const IModelFunction & func, const BinData
       else {
          SimpleGradientCalculator  gc( npar, func);
          if (!useBinIntegral ) {
-            gc.ParameterGradient(x, p, fval, g);
+            // need to use un-normalized fval
+            gc.ParameterGradient(x, p, fval0, g);
```

Comment:
Because the function `ParameterGradient` computes the gradient by calculation f(x+delta) -f(x). f(x) is not recalculated to save time and eval is used. The function does not know about normalisation so one needs to use the uncorrected value. The gradient is then normalised by the bin volume later in the code (line 651)